### PR TITLE
#167313423 highlight text in the article

### PR DESCRIPTION
**Pull request by habinezadalvan**
#### What does this PR do?

- Highlights text the article

#### Description of Task to be completed?

- Completed the functionality where a user can highlight, comment on a highlight, share a highlight on different social platforms like facebook, twitter, LinkedIn, Gmail.

- Completed testing functions are tested at 100%

- The completed API documentation of all endpoints 

#### How should this be manually tested?

- Download or clone the tutorial project code from this repo, on you local terminal checkout to ft-highlight-167313423.

- install all required NPM packages by running **npm ci** 
- Add all environment variables found in **.env.sample** in you **.env** file located outside the src folder.
- run **npm test**  or **npm run dev** to verify if everything is working well. 

- To create table and seeds use: **npm run migration** and **npm run seeds**

- If everything is working then go to you postman and use the following URLs. 

**To highlight something in an article**
- Method: POST: http://localhost:3000/api/v1/articles/:articleslug/highlight?startIndex=100&&endIndex=20
**To get all the highlights of a particular article**
- Method: GET: http://localhost:3000/api/v1/articles/:articleId/highlight
**Delete a highlight comment**
- Method: DELETE: http://localhost:3000/api/v1/articles/highlight/:highlightId
**To share a highlight to different platforms**
- Method: GET: http://localhost:3000/api/v1/articles/:highlightId/highlight/share/:channel
> example of channels: facebook, linkedin, twitter

#### Any background context you want to provide?

- used Sequelize framework to implement the functionality

#### What are the relevant pivotal tracker stories?
[167313423](https://www.pivotaltracker.com/story/show/167313423)
#### Screenshots (if appropriate)
Screen Shot 2019-08-28 at 18 49 46
Screen Shot 2019-08-28 at 14 37 54
Screen Shot 2019-08-28 at 14 38 30
Screen Shot 2019-08-28 at 14 39 25
Screen Shot 2019-08-28 at 14 40 42
Screen Shot 2019-08-28 at 14 41 26
Screen Shot 2019-08-28 at 15 50 23
Screen Shot 2019-08-28 at 14 38 09
Screen Shot 2019-08-28 at 18 31 17
Screen Shot 2019-08-28 at 18 38 44
Screen Shot 2019-08-28 at 18 43 00
Screen Shot 2019-08-28 at 18 47 51

#### Questions:
none

### CODE REVIEW

File: src/middlewares/shareHighlight.middleware.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+export default async (req, res, next) => {
+  const { id } = req.params;
+  const hightlight = await Helper.findHighlight(id);
+  if ((hightlight === null) || (hightlight.length === 0)) {
+    util.setError(404, 'Sorry you can not share a highlight that does not exist');
+    return util.send(res);
+  }
+  const [{ text }, articleSlug] = [hightlight, hightlight.get().highlight.get().slug];
+  if (req.url.search(/\/facebook/g) > 0) {
```
Comment (nkalyesubula) [inline src/middlewares/shareHighlight.middleware.js:19]: A switch statement would have been a better option

File: test/search.test.js [hunk 2]
```diff
@@ -1,50 +1,50 @@
...
-//   verified: false
-// });
+const usertoken = Helper.generateToken({
+  id: 2,
+  email: 'user@gmail.com',
+  username: 'user',
+  verified: false
+});
 
-// describe('search article query builder', () => {
```
Comment (nkalyesubula) [inline test/search.test.js]: Fix these tests please

File: test/test.articles.js [hunk 3]
```diff
@@ -49,24 +49,24 @@ describe('Articles', () => {
     await articleController.createArticles(req, res);
     expect(res.status).to.have.been.calledWith(201);
   });
-  // it('List of all articles', (done) => {
```
Comment (nkalyesubula) [inline test/test.articles.js]: the same here

File: test/highlight.test.js [hunk 4]
```diff
@@ -0,0 +1,274 @@
+import sinon from 'sinon';
+import { chai, server, expect } from './test-setup';
+import Helper from '../src/helpers/helper';
+import OpenUrlHelper from '../src/helpers/share.article.helper';
+import highlightController from '../src/controllers/highlight.controller';
+import shareMiddleware from '../src/middlewares/shareHighlight.middleware';
+import commentsService from '../src/services/comments.service';
+
+chai.should();
```
Comment (nkalyesubula) [inline test/highlight.test.js]: This code is not used. Could use remove it please

**Review by nkalyesubula — DISMISSED** [review]
LGTM

**Review by Mnickii — DISMISSED** [review]
LGTM

**Pull request merged**